### ISSUE TITLE
clang-tidy: hicpp-use-emplace: False positive with incorrect fix

### ISSUE THREAD

**Issue by kees-jan** [llvm/llvm-project#62387]
Consider the following code:
``` C++
#include <vector>

struct S
{
    int i;
    std::vector<int> is;
};

void f()
{
    std::vector<S> s;
    s.push_back({1,{}});
}
```
Check `hicpp-use-emplace` will suggest to use emplace:
``` console
[<source>:12:7: warning: use emplace_back instead of push_back [hicpp-use-emplace]](javascript:;)
    s.push_back({1,{}});
      ^~~~~~~~~~~~~~~~
      emplace_back(
1 warning generated.
```
This, to the best of my knowledge, is incorrect. `emplace_back` cannot be used to initialize a struct. [StackOverflow seems to concur](https://stackoverflow.com/questions/20603335/c11-emplace-back-and-push-back-syntax-with-struct).

The suggested fix is to replace that line with `s.emplace_back()` (no arguments). This is definitely incorrect. It results in a default constructed struct, as opposed to one matching the provided initializer.

[See this example on compiler explorer](https://godbolt.org/z/fb3eY8Gs9)
